R/smooth_mortality_law.R

```diff
@@ -0,0 +1,90 @@
+#' Function to smooth mortality rates using using MortalityLaw package
+#'
+#' This smoothing function allows smoothing of a variable in a vital object using MortalityLaw package.
+#' The vital object is returned along with some additional columns containing
+#' information about the smoothed variable: usually `.smooth` containing the
+#' smoothed values, and `.smooth_se` containing the corresponding standard errors.
+#'
+#' `smooth_mortality_law()` smoothes mortality rates using specified mortality law and optimisation method.
+#' @param .data A vital object
+#' @param .var name of variable to smooth
+#' @param law name of mortality law. For available mortality laws, users can check the \code{\link{MortalityLaws::availableLaws}}
+#' function to learn about the available options.
+#' @param opt.method optimisation method. The models can be fitted under the maximum likelihood methodology
+#' or by selecting a loss function to be optimised. See the implemented loss function by running the \code{\link{MortalityLaws::xavailableLF}} function.
+#' @return vital with added columns containing smoothed values and their standard errors
+#' @keywords smooth, mortality
+#' @rdname smooth_law_vital
+#' @author Sixian Tang and Rob J Hyndman
+#' @examples
+#' library(MortalityLaws)
+#' norway_mortality |> smooth_mortality_law(Mortality)
+#'
+#' @export
+smooth_mortality_law <- function(.data, .var, law = "gompertz", opt.method = "LF2") {
```

Comment:
Is gompertz the best default?

The `MortalityLaw()` function allows custom laws using simple functions. You could allow for this by adding a `...` argument in `smooth_mortality_law()` and passing it through to `MortalityLaw()`. Then there would be no need for an `opt.method` argument in our function, because it could also be passed through via `...`.

I don't think you need a `smooth_law_vital()` function. You can just use the existing `smooth_vital()` function and pass through the arguments as necessary.



It would be good to also pass through `Dx` and `Ex` if they exist in the data set. You can find these using